### PR TITLE
Feature/add calls

### PR DESCRIPTION
**Pull request by sergeyevalex90**
Add socket service for calls on backend
Add socket service for calls on frontend
Add call component on frontend
Implement React Jitsi SDK
Create stopwatch component

### CODE REVIEW

File: front-end/package.json [hunk 1]
```diff
@@ -44,6 +45,7 @@
     "socket.io-client": "^4.5.4",
     "suneditor": "^2.44.3",
     "suneditor-react": "^3.4.1",
+    "use-sound": "^4.0.1",
     "web-vitals": "^2.1.4"
   },
   "scripts": {
```
Comment (andriydzhrvts) [inline front-end/package.json:51]: Do you need iphone image?

File: front-end/src/components/common/RingingPhone/RingingPhone.jsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+          className={
+            active ? `${styles.callAnimation} ${styles.withoutAnimation}` : styles.callAnimation
+          }
+        >
+          <img src={role === TEACHER_ROLE ? student.url : teacher.url} alt='' width='135' />
+        </div>
+        <div className={styles.phoneBtns}>
+          {active && !isCallingUser && !onLoading && !onJoining && (
+            <div className={styles.phoneBtn} onClick={callApproveHandler}>
+              <svg height='24' viewBox='0 0 24 24' width='24' xmlns='http://www.w3.org/2000/svg'>
```
Comment (andriydzhrvts) [inline front-end/src/components/common/RingingPhone/RingingPhone.jsx]: Replace to assets

File: front-end/src/components/common/RingingPhone/RingingPhone.jsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                <path d='M0 0h24v24H0z' fill='none' />
+                <path d='M6.62 10.79c1.44 2.83 3.76 5.14 6.59 6.59l2.2-2.2c.27-.27.67-.36 1.02-.24 1.12.37 2.33.57 3.57.57.55 0 1 .45 1 1V20c0 .55-.45 1-1 1-9.39 0-17-7.61-17-17 0-.55.45-1 1-1h3.5c.55 0 1 .45 1 1 0 1.25.2 2.45.57 3.57.11.35.03.74-.25 1.02l-2.2 2.2z' />
+              </svg>
+            </div>
+          )}
+          {onLoading && !onJoining && <DotsSpinner />}
+          {isCallingUser && !onJoining && <DotsSpinner />}
+          {onJoining && <Timer running={timer} />}
+          <div className={`${styles.phoneBtn} ${styles.declineCall}`} onClick={declineCallHandler}>
+            <svg height='24' viewBox='0 0 24 24' width='24' xmlns='http://www.w3.org/2000/svg'>
```
Comment (andriydzhrvts) [inline front-end/src/components/common/RingingPhone/RingingPhone.jsx]: The same as ^

File: front-end/src/components/generalLayout/GeneralLayout.jsx [hunk 4]
```diff
@@ -2,6 +2,7 @@ import React from 'react';
 import styles from './GeneralLayout.module.scss';
 import Sidebar from 'components/Sidebar/Sidebar';
 import { Outlet } from 'react-router';
+//import MeetRoom from 'components/MeetRoom/MeetRoom';
```
Comment (andriydzhrvts) [inline front-end/src/components/generalLayout/GeneralLayout.jsx]: Remove commented line

**Review by andriydzhrvts — APPROVED** [review]
Good work! Approve after resolving comments

File: front-end/src/components/Timer/Timer.jsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+// Styles
+import styles from './Timer.module.scss';
+
+const Timer = ({ running }) => {
+  const [time, setTime] = useState(0);
+
+  useEffect(() => {
+    let interval;
+    if (running) {
```
Comment (andriydzhrvts) [inline front-end/src/components/Timer/Timer.jsx:12]: Rewrite to `if (running) {
      interval = setInterval(() => {
        setTime((prevTime) => prevTime + 10);
      }, 10);
return;
    } 
      clearInterval(interval);
    `

File: front-end/package.json (hunk 1 above)
Comment (sergeyevalex90) [reply front-end/package.json:51]: Removed

File: front-end/src/components/common/RingingPhone/RingingPhone.jsx (hunk 2 above)
Comment (sergeyevalex90) [reply front-end/src/components/common/RingingPhone/RingingPhone.jsx]: Done

File: front-end/src/components/common/RingingPhone/RingingPhone.jsx (hunk 3 above)
Comment (sergeyevalex90) [reply front-end/src/components/common/RingingPhone/RingingPhone.jsx]: Done

File: front-end/src/components/generalLayout/GeneralLayout.jsx (hunk 4 above)
Comment (sergeyevalex90) [reply front-end/src/components/generalLayout/GeneralLayout.jsx]: Done

File: front-end/src/components/Timer/Timer.jsx (hunk 5 above)
Comment (sergeyevalex90) [reply front-end/src/components/Timer/Timer.jsx:12]: Done

**Pull request merged**